contracts/core/Factory.sol

```diff
@@ -51,7 +52,12 @@ contract Factory is IFactory {
       token1 = token0;
       token0 = WETH;
     }
+
+    // what's the reason for not providing all these parameters in the constructor?
+    // But at the same time, the pattern of assigning a global var, read then delete is interesting
+    // It's the first time I see this pattern
```

Comment:
As I know, if the constructor has arguments, then the address of the contract will not be deterministic

Reply:
I have posted an explaination in the server, and I saw you already reacted to it lol